plotly/figure_factory/_dendrogram.py

```diff
@@ -90,14 +90,15 @@ def create_dendrogram(X, orientation="bottom", labels=None,
                              distfun=distfun, linkagefun=linkagefun,
                              hovertext=hovertext)
 
-    return graph_objs.Figure(data=dendrogram.data, layout=dendrogram.layout)
+    return graph_objs.Figure(data=dendrogram.data,
+                             layout=dendrogram.layout)
 
 
 class _Dendrogram(object):
     """Refer to FigureFactory.create_dendrogram() for docstring."""
 
     def __init__(self, X, orientation='bottom', labels=None, colorscale=None,
-                 width="100%", height="100%", xaxis='xaxis', yaxis='yaxis',
+                 width=np.inf, height=np.inf, xaxis='xaxis', yaxis='yaxis',
```

Comment:
this seems odd to me: infinite width?

Reply:
Context: The string '100%' is forbidden by the number `valType` of `width`.

Reply:
Got it. I wonder what `width="100%"` was doing before. I don't think that it doesn't anything in plotly.js as `width='100%'` and `height='100%'` should correspond to `autosize=True`.

Maybe the better setting would be to:
- set the arguments as `width=None` and `height=None` 
- if `width` or `height` is not `None`, apply it in the figure
- set `autosize=True`